src/scripting/game_lua_kernel.cpp

```diff
@@ -2750,6 +2750,44 @@ int game_lua_kernel::intf_set_floating_label(lua_State* L, bool spawn)
 	return 1;
 }
 
+namespace
+{
+	const unit_map& get_unit_map()
+	{
+		// Used if we're in the game, including during the construction of the display_context
+		if(resources::gameboard) {
+			return resources::gameboard->units();
+		}
+
+		// If we get here, we're in the scenario editor
+		assert(display::get_singleton());
+		return display::get_singleton()->context().units();
+	}
+	void reset_affect_adjacent(const unit& u_)
+	{
+		bool affect_adjacent = false;
+		for(const auto [key, cfg] : u_.abilities().all_children_view()) {
+			bool image_or_hides = (key == "hides" || cfg.has_attribute("halo_image") || cfg.has_attribute("overlay_image"));
+			if(image_or_hides && cfg.has_child("affect_adjacent")){
+				affect_adjacent = true;
+				break;
+			}
+		}
+		const unit_map& units = get_unit_map();
+		if(affect_adjacent) {
+			const auto adjacent = get_adjacent_tiles(u_.get_location());
+			for(unsigned i = 0; i < adjacent.size(); ++i) {
+				const unit_map::const_iterator it = units.find(adjacent[i]);
+				if (it == units.end() || it->incapacitated())
+					continue;
+				if ( &*it == &u_ )
```

Comment:
This checks if the unit is adjacent to itself?

Reply:
if unit checked is itself, no need to refresh itself

Reply:
How is a unit adjacent to itself?

Reply:
> How is a unit adjacent to itself?

This precaution also exists in get_abilities get_ability_bool to avoid that the central unit is checked with its adjacent units, I only continue to apply this principle

Reply:
There is also a big comment there explaining why. How is that relevant here?

Reply:
if the unit possessing the ability sees its unit sprite updated while moving, the movement animation will be replaced by the standing_anim, (same for die animation etc..) this is why you should not check the unit owning the ability with the image_halo= attribute coupled with [affect_adjacent)

Reply:
The comment I'm referring to:
https://github.com/wesnoth/wesnoth/blob/bcfda40c4981c7fd2a3bc74414bef3b08b01d2a3/src/units/abilities.cpp#L202-L206

How does that relate to this function?

Reply:
@soliton- I rewrote the comment to clarify that it is an assertion that the unit checked is always the unit adjacent to the unit playing the animation and not the unit itself, even though the latter is highly unlikely, if not impossible.

Reply:
@soliton- i removed the checking itself